### DISCUSSION TITLE
Is there a way to get the HTML/JSON value of a text range?

### DISCUSSION THREAD

**Discussion by kawaicheung** [ueberdosis/tiptap#1953] [General Questions] [answered]
I'm trying to figure out a way to grab the part of the text that is after my current cursor position and move all of that text to a newly created text editor below it.

Leveraging ProseMirror, I can grab the necessary content like so:

`this.editor.state.selection.content().toString()` or `....toJSON()`

However, when instantiating a new tiptap editor, I can't pass either of these formats into the `content` property as neither format is purely HTML or in the same JSON format that tiptap expects.

Is there a tiptap native method to grab the HTML or JSON data of a range of text so that I can pass that into the instantiation of another editor?

**Comment by philippkuehn** [comment, marked as answer]
Hey, it’s because `toJSON()` returns a fragment. It’s possible that this fragment is not compatible as part of a new document (e.g. that fragment contains only text and your doc schema expects block nodes at to level).

One possible solution is to convert that fragment to an HTML string based on your schema. A text fragment would be wrapped by a paragraph (or something else defined in your schema) for example. That HTML string can easily parsed as a new document then.

```js

import { getHTMLFromFragment } from '@tiptap/core'

const selectionHTML = getHTMLFromFragment(editor.state.selection.content(), editor.schema)
```

Please make sure you’ve updated to the latest package versions because I added [this patch](https://github.com/ueberdosis/tiptap/commit/2c9cf03398e2b14fceaec4386fdb53cae5261ae7) to `getHTMLFromFragment`.

**Reply by kawaicheung** [reply]
Thanks @philippkuehn - that worked like a charm!

One correction from your code. I had to grab the `content` property (the `Fragment`) from `.content()` which is a `Slice`. I'm using the vue-2 version so not sure if the schema changed between them. 

`const selectionHTML = getHTMLFromFragment(editor.state.selection.content().content, editor.schema)`
